src/mt_protocol.cpp

```diff
@@ -138,10 +141,365 @@ bool mt_send_heartbeat() {
 
 }
 
+void set_portnum_callback(void (*callback)(uint32_t from, uint32_t to,  uint8_t channel, meshtastic_PortNum port, meshtastic_Data_payload_t *payload)) {
+  portnum_callback = callback;
+}
+
+void set_encrypted_callback(void (*callback)(uint32_t from, uint32_t to,  uint8_t channel, meshtastic_MeshPacket_public_key_t pubKey, meshtastic_MeshPacket_encrypted_t *payload)) {
+  encrypted_callback = callback;
+}
+
 void set_text_message_callback(void (*callback)(uint32_t from, uint32_t to,  uint8_t channel, const char* text)) {
   text_message_callback = callback;
 }
 
+bool handle_id_tag(uint32_t id) {
+  Serial.printf("id_tag: ID: %d\r\n", id);
+  return true;
+}
+
+bool handle_config_tag(meshtastic_Config *config) {
+  switch (config->which_payload_variant) {
+    case meshtastic_Config_device_tag:
+      Serial.printf("Config:device_tag:  role: %d\r\n", config->payload_variant.device.role);
```

Comment:
Not sure it's a good idea to print every single entry in a function of the library itself. It creates a wall of text and you cannot disable it unless you modify the library. These entries also may change, and new ones may be added in the future. 
If you want to print something, I would just only print which tag it is, and only when `mt_debugging` is enabled.